### ISSUE TITLE
EpicFight compat

### ISSUE THREAD

**Issue by Illest8** [BeansGalaxy/Beans-Backpacks-2#154]
![2024-05-24_23 20 42](https://github.com/BeansGalaxy/Beans-Backpacks-2/assets/94665911/ea84f948-7861-404e-82df-058e66b35783)

I was hoping that you could manipulate the curios renderer to allow backpacks to be worn like the Quark backpack shown above. This fixes a lot of issues with mods that change any upper body animations (i.e. EpicFight). Quark is able to do this because the backpack is worn in the chest slot but it limits players able to wear a chestplate simultaneously. I was wondering if there's a way to make it so that the backpack renderer is connected to the chest slot whilst being a separate back slot so that the backpack is rendered on the back of the player torso?

Thank you!

![2024-05-24_23 21 11](https://github.com/BeansGalaxy/Beans-Backpacks-2/assets/94665911/e1b0f256-6308-468b-b683-38fb4c9b078f)

**Comment by BeansGalaxy** [comment]
I'll check why the backpack is disconnected from the back with those mods. Are you also asking if you could equip my backpacks in the chestplate?

**Comment by Illest8** [comment]
Not exactly. I observed that Quark used the chest plate slot to have the backpack stay attached to the torso regardless of the mod that adds animations. My thought process is that you could make it so the backpack still goes in the curios backslot but renders on the back of the torso like the screenshot I took of wearing the Quark backpack.

**Comment by BeansGalaxy** [comment]
Ah okay Ive got it now. You don't care where the backpack is stored, you don't want it off the back.

Do you have anything equipped in the chestplate? The backpack renders a bit off the back so it sits on top of armor but it just checks if the chestplate isn't empty, not if the chestplate's model actually would interfere with it.

**Comment by Illest8** [comment]
With the Quark backpack it's actually ONLY equippable in the chest slot, therefore a chest plate can't be worn simultaneously. Perhaps that's why it works so well with the EpicFight and PlayerAnimations mods. I'm not entirely familiar with how the renderers work so if it's not possible to keep the backpack attached to the torso at the same time as a chest plate without clipping then I will close the issue.

**Comment by BeansGalaxy** [comment]
Currently racing the 1.21 release to try to be the first backpack mod out but I haven't forgot this issue :)
